examples/creating-cert-chain-with-validation.cpp

```diff
@@ -0,0 +1,247 @@
+            .addExtension(intermediateCaSignConstraints)
+            .build();
+
+    return CertificateAuthority(
+                intermediateCaSignParams,
+                // first serial number of the new certificates this CA signs
+                1000000,
+                intermediateCaCert,
+                intermediateCaKey);
+}
+
+/*
+* Sign (end Cert) certificate with intermediate CA based on given CSR
+*/
+X509Certificate signCertificateWithIntermediateCa(const X509Certificate &signingCert,
+                                                  const AsymmetricKeypair &signingKey,
+                                                  const CertificateSigningRequest &csr)
+{
+    auto ca = defineIntermediateCa(signingCert, signingKey);
+    try {
+        return ca.signCSR(csr);
+    } catch (const openssl::OpenSSLException &e) {
+        /* low level OpenSSL failure */
+        std::cerr << "Error reading CSR in PEM format. Please validate your CSR." << std::endl;
+        std::cerr << e.what() << std::endl;
+        exit(EXIT_FAILURE);
+    } catch (const MoCOCrWException &e) {
+        /* Reason:
+         * - sanity checks of the certificate failed */
+        std::cerr << "Failed to create certificate." << std::endl;
+        std::cerr << e.what() << std::endl;
+        exit(EXIT_FAILURE);
+    }
+}
+
+int main()
+{
+    // Make root CA
+    auto rootCaKeypair = AsymmetricKeypair::generateRSA();
+    auto rootCaCert = createRootCaCertificate(rootCaKeypair);
+
+    // Make intermediate CA
+    auto intermediateCaEccKey = AsymmetricKeypair::generateECC();
+    auto intermediateCaCert = createIntermediateCaCertificate(intermediateCaEccKey, rootCaKeypair, rootCaCert);
+
+    // Sign a certificate with intermediate CA
+    auto endCertKeypair = AsymmetricKeypair::generateRSA();
+    auto endCertDetails = DistinguishedName::Builder{}
+            .organizationalUnitName("Car IT")
+            .organizationName("BMW")
+            .countryName("DE")
+            .commonName("MyCert")
+            .build();
+    auto csr = CertificateSigningRequest(endCertDetails,
+                                         endCertKeypair,
+                                         digestType);
+    auto endCert = signCertificateWithIntermediateCa(intermediateCaCert, intermediateCaEccKey, csr);
+
+    // Verify end certificate with intermediate CA's certificate
+    std::cout << "Verifying end certificate...\n";
+    try {
+        endCert.verify({rootCaCert}, {intermediateCaCert});
+    } catch (const MoCOCrWException &e) {
+        std::cerr << "Failed to verify end certificate.\n";
+        std::cerr << e.what() << '\n';
+        exit(EXIT_FAILURE);
+    }
+    std::cout << "Success!\n";
+
+    // Verify after end certificate expiry
+    std::cout << "Verifying after end certificate expired...\n";
+    std::this_thread::sleep_for(std::chrono::seconds(6)); // Intermediate CA signs certificate that last 5 seconds
```

Comment:
Well, let's try to avoid weird racy not best practice code in the examples. We have seen in particular that this short expiry times are problematic when running in emulators (e.g. ARM). Better make the validity times longer and illustrate time checks with the ability to check at a given point in time.